src/edu/csus/ecs/pc2/core/model/Problem.java

```diff
@@ -141,18 +183,33 @@
          */
         FAILED, 
         /**
-         * No Input Validator has been run on the data files in this problem.
-         */
-        NOT_TESTED,
-        /**
-         * An attempt was made to run an Input Validator but an internal error occurred.
+         * An internal error occurred. One reason for this could be that an attempt was made to run an Input Validator
+         * but it failed.
          */
         ERROR
     }
     
-    private InputValidationStatus inputValidationStatus;
+    /**
+     * The type of Input Validator currently associated with the problem.
+     */
+    private INPUT_VALIDATOR_TYPE currentInputValidatorType = INPUT_VALIDATOR_TYPE.NONE;
     
-    private Vector<InputValidationResult> inputValidationResults ;
+    //custom input validator settings - only relevant if the problem was saved with a Custom Input Validator
+    private boolean problemHasCustomInputValidator = false;
+    private boolean customInputValidatorHasBeenRun = false;
+//    private String customInputValidatorProgramName = "";  //program name should be determined by the Custom Input Validator Serialized File
+    private String customInputValidatorCommandLine = "";
+//    private String customInputValidatorFilesOnDiskFolderName = "";   
+    private SerializedFile customInputValidatorSerializedFile = null;
+    private InputValidationStatus customInputValidationStatus = InputValidationStatus.UNKNOWN;
+    private Vector<InputValidationResult> customInputValidationResults = null;
+
+    //VIVA Input validator settings - only relevant if the problem was saved with Viva Input Validator Settings
+    private boolean problemHasVivaInputValidatorPattern = false;
+    private boolean vivaInputValidatorHasBeenRun = false;
+    private String [] vivaInputValidatorPattern = null;
+    private InputValidationStatus vivaInputValidationStatus = InputValidationStatus.UNKNOWN;
+    private Vector<InputValidationResult> vivaInputValidationResults = null;
```

Comment:
Instead of ProblemDataFile, put VIVA settings in VivaInputValidatorSettings.   VIVA settings will be stored
in Problem, until a refactor moves their storage outside of Problem similar to how ProblemDataFiles are stored.

Reply:
With move of all Viva settings into new VivaInputValidatorSettings class, this item is resolved